src/services/auth/sponsor/sponsor-router.ts

```diff
@@ -0,0 +1,118 @@
+import { Router } from "express";
+import { Database } from "../../../database";
+import RoleChecker from "../../../middleware/role-checker";
+import { Role } from "../auth-models";
+import { StatusCodes } from "http-status-codes";
+import { sendEmail } from "../../ses/ses-utils";
+import jsonwebtoken from "jsonwebtoken";
+import { Config } from "../../../config";
+import * as bcrypt from "bcrypt";
+const sponsorRouter = Router();
+
+// Get favorite events for an attendee
+sponsorRouter.get(
+    "/",
+    RoleChecker([Role.Enum.CORPORATE]),
+    async (req, res, next) => {
+        try {
+            const resumeUsers = await Database.REGISTRATION.find(
+                { hasResume: true },
+                { userId: 1 }
+            );
+            if (!resumeUsers) {
+                return res
+                    .status(StatusCodes.NOT_FOUND)
+                    .json({ error: "UserNotFound" });
+            }
+            return res.status(StatusCodes.OK).json(resumeUsers);
+        } catch (error) {
+            next(error);
+        }
+    }
+);
+
+function createSixDigitCode() {
+    let result = "";
+    const chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+    for (let i = 0; i < 6; i++) {
+        result += chars.charAt(Math.floor(Math.random() * chars.length));
+    }
+    return result;
+}
+
+function encryptSixDigitCode(sixDigitCode: string): string {
+    console.log("SixDigit: ", sixDigitCode);
+    const saltRounds = 10;
+
+    try {
+        const hash = bcrypt.hashSync(sixDigitCode, saltRounds);
+        return hash;
+    } catch (err) {
+        console.error("Error encrypting the code:", err);
+        throw err;
+    }
+}
+
+sponsorRouter.post("/login", async (req, res, next) => {
+    const { email } = req.body;
+    try {
+        const sixDigitCode = createSixDigitCode();
+        const expTime = Math.floor(Date.now() / 1000) + 300;
+        const hashedVerificationCode = encryptSixDigitCode(sixDigitCode);
+        await Database.AUTH_CODES.findOneAndUpdate(
+            { email },
+            {
+                $set: {
+                    hashedVerificationCode: hashedVerificationCode,
+                    expTime: expTime,
+                },
+            },
+            { upsert: true }
+        );
+        await sendEmail(
+            email,
+            "RP-Sponor Email Verification!",
+            `Here is your verification code: ${sixDigitCode}`
+        );
+        return res.sendStatus(StatusCodes.CREATED);
+    } catch (error) {
+        next(error);
+    }
+});
+
+sponsorRouter.post("/verify", async (req, res, next) => {
+    const { email, sixDigitCode } = req.body;
+    try {
+        const sponsorData = await Database.AUTH_CODES.findOne({ email });
```

Comment:
dont do find and then delete. await delete here, and use the returned object for comparisons

Reply:
https://mongoosejs.com/docs/api/model.html#Model.deleteOne()

Reply:
ahh thanks